samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -128,11 +128,11 @@ class KafkaCheckpointManager(checkpointSpec: KafkaStreamSpec,
     info(s"Reading checkpoint for taskName $taskName")
 
     if (taskNamesToCheckpoints == null) {
-      debug("Reading checkpoints for the first time")
+      info("Reading checkpoints for the first time")
       taskNamesToCheckpoints = readCheckpoints()
-    } else {
-      debug("Updating existing checkpoint mappings")
-      taskNamesToCheckpoints ++= readCheckpoints()
+      // Stop the system consumer since we only need to read checkpoints once
```

Comment:
Don't need the else anymore?

If not, does it make sense to move the register and start here as well? Maybe even make it local to the method instead of a field? Is it  used for anything else?

Reply:
'it' == systemConsumer. That way we won't break anything if something does rely on reading multiple times, and will still get the perf improvement for the happy path where we don't.

Reply:
No, i don't think that update is needed at all. It's weird after the first pulling we are still trying to update. It doesn't make it better if there is any problem in the first read (some task also has wrong checkpoints).

I feel it might be useful to start a little bit early to just give a bit more time to let the proxy thread running. It's not used anywhere else.

Reply:
Them maybe leave a comment near the start that this is stopped after first checkpoint read. Otherwise LGTM, nice find.

Reply:
cool! Let me put a comment there.